pravega/pravega
# Issue 4054: Make common host tag consistent and shorter across controller and segmentstore

Signed-off-by: kevinhan88 <kevinhan88@gmail.com>

**Change log description**  
This change checks if "HOSTNAME" is defined as system property (in the case of docker deployment it is) and use the defined hostname for the common host tag, in order to:
- have consistent host tag across controllers and segmentstores, and across pravega metrics and system metrics. This will make the metrics dashboard queries consistent and simpler.
- be flexible for the set of hostname. In this particular case, shorter hostname without the repetitive tailing segments.

**Purpose of the change**  
Fixes #4054, #3923 

**What the code does**  
When metrics is being initialized, instead of go straight to call InetAddress.getLocalHost().getHostName(), the code checks if "HOSTNAME" is defined and use it if so.

**How to verify it**  
Additional unit test added.
On metrics dashboard, consistent (and shorter) hostname should show up as host tag.
If we want to change the hostname for some reason, we can do so by changing system var, and then the metrics should reflect the change.
